### DISCUSSION TITLE
Example for accessing GitLab container registry

### DISCUSSION THREAD

**Discussion by Guite** [renovatebot/renovate#18856] [Request Help]
Hello,

I am trying to fix the `Failed to look up docker dependency registry.gitlab.com/xxx/yyy/zzz` messages.

In [this discussion](https://github.com/renovatebot/renovate/discussions/17186#discussioncomment-3403752) I found the hint that I need to add a host rule for the GitLab registry. Hence, I've added this to `renovate.json5`:

```
  hostRules: [
    {
      matchHost: 'registry.gitlab.com',
      username: 'gitlab-ci-token',
      password: '$RENOVATE_REGISTRY_TOKEN',
    },
  ],
```

In the `renovate-runner` repository I added a `RENOVATE_REGISTRY_TOKEN` variable in the CI/CD settings.
Seems I am still missing something.

Thanks in advance for any pointers.

**Comment by viceice** [comment]
don't prefix custom env vars with `RENOVATE_`, renovate tries to interpret those, so it can cuase issues.

also renovate don't replace any env vars inside json config files. 

https://docs.renovatebot.com/getting-started/private-packages/

**Reply by Guite** [reply]
Thank you.

**Reply by felixranesberger** [reply]
Hi, I don't fully understand the provided solution.

When installing from the Gitlab package registry you normaly can use the `${CI_JOB_TOKEN}` to authenticate yourself. 

`npm config set -- '//gitlab.domain.de/api/v4/projects/321/packages/npm/:_authToken' "${CI_JOB_TOKEN}"`

As you already stated above, you can't use the environment variables directly inside the `renovate.json` file. 

What do I have to do to allow a injection of the token inside the token string?

I've seen some [examples](https://docs.renovatebot.com/getting-started/private-packages/#add-hostrule-to-bots-config) using module.exports and process variables, 
but I'm not sure, in what combination these solutions are used.

Could you help me?
